### PR TITLE
[dvsim] update proj_root logic

### PR DESCRIPTION
**Pull request by udinator**
Add a proj_root command line variable to dvsim
Update the get_proj_root function to get the root of the current git repo

Signed-off-by: Udi Jonnalagadda <udij@google.com>

### CODE REVIEW

File: util/dvsim.py [hunk 1]
```diff
@@ -81,7 +81,14 @@ def resolve_branch(arg_branch):
 
 # Get the project root directory path - this is used to construct the full paths
 def get_proj_root():
-    return os.path.abspath(os.path.dirname(os.path.realpath(__file__)) + "/..")
+    result = subprocess.run(["git", "rev-parse", "--show-toplevel"],
+                             stdout=subprocess.PIPE)
```
Comment (sriyerg) [inline util/dvsim.py]:
```suggestion
                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

File: util/dvsim.py [hunk 2]
```diff
@@ -81,7 +81,14 @@ def resolve_branch(arg_branch):
...
+    proj_root = result.stdout.decode("utf-8").strip()
+    if proj_root == "":
+        log.error(
+            "Failed to find proj_root.")
```
Comment (sriyerg) [inline util/dvsim.py]: Perhaps provide more context: `("Attempted to find the root directory of this GitHub repository by running:\n".." \nBut the command appears to have failed: \n%s", result.stderr.decode("utf-8")`

File: util/dvsim.py [hunk 3]
```diff
@@ -127,6 +134,17 @@ def main():
...
 
+    parser.add_argument(
+        "-pr",
+        "--proj-root",
+        metavar="path",
+        help="""root scratch directory path where all build / run drectories go;
+                      by default, the tool will create the {scratch_path} = {scratch_root}/{dut}
+                      directory if it doesn't already exist; under {scratch_path}, there will be
+                      {compile_set} set of directories where all the build outputs go and
+                      {test_name} set of directories where the test outputs go"""
```
Comment (sriyerg) [inline util/dvsim.py]: Update this description.
"Specify the root directory of the project. If this option is not passed, the tool will assume that this is a local GitHub repository and will attempt to automatically find it."

File: util/dvsim.py [hunk 4]
```diff
@@ -445,6 +463,11 @@ def main():
     # be deployed.
     # TODO: SimCfg item below implies DV - need to solve this once we add FPV
     # and other ASIC flow targets.
+    if args.proj_root:
```
Comment (sriyerg) [inline util/dvsim.py]: Add newline with an appropriate comment.

File: util/dvsim.py (hunk 1 above)
Comment (udinator) [reply util/dvsim.py]: Done

File: util/dvsim.py (hunk 2 above)
Comment (udinator) [reply util/dvsim.py]: Done, thanks for the suggestion

File: util/dvsim.py (hunk 4 above)
Comment (udinator) [reply util/dvsim.py]: Done.

File: util/dvsim.py (hunk 3 above)
Comment (udinator) [reply util/dvsim.py]: Done

**Pull request merged**